pkg/cluster/internal/providers/provider.go

```diff
@@ -32,6 +32,9 @@ type Provider interface {
 	// ListClusters discovers the clusters that currently have resources
 	// under this providers
 	ListClusters() ([]string, error)
+	// ListAllNodes discovers the nodes that currently have resources
+	// under this providers
+	ListAllNodes() ([]nodes.Node, error)
```

Comment:
I think we should just call list clusters and then call list nodes for each cluster.

Sure it's a bit slower, but this method has a single actual use case, complicating adding more providers for one niche command. Users should not have a super high number of clusters / nodes realistically anyhow, and they can always reach around and use ps themselves.